test/test_Radar.cpp

```diff
@@ -54,4 +56,17 @@ TEST_F(RadarTest, it_crashes_with_different_range)
     radar.addEcho(range, sweep_length, step_size, heading_0, echo);
     ASSERT_THROW(radar.addEcho(range + 2.0, sweep_length, step_size, heading_1, echo),
         std::runtime_error);
+}
+
+TEST_F(RadarTest, verify_next_angle_works_as_expected)
```

Comment:
"works as expected" is not a specification. You need to split your tests and say exactly what behavior you are testing against.

TEST_F(..., it_returns_true_for_the_first_echo) {
}

TEST_F(..., it_returns_true_if_the_angle_is_within_size_times_step_from_the_first_with_a_positive_step_angle) {
}

TEST_F(..., it_returns_true_if_the_angle_is_within_size_times_step_from_the_first_with_a_negative_step_angle) {
}
```